go-controller/pkg/clustermanager/networkconnect/hybrid_connect_subnet_allocator_test.go

```diff
@@ -0,0 +1,1547 @@
+package networkconnect
+
+import (
+	"fmt"
+	"net"
+	"testing"
+
+	"github.com/onsi/gomega"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+)
+
+func mustParseCIDR(cidr string) *net.IPNet {
+	_, network, err := net.ParseCIDR(cidr)
+	if err != nil {
+		panic(err)
+	}
+	return network
+}
+
+func TestHybridConnectSubnetAllocator_AddNetworkRange(t *testing.T) {
+	tests := []struct {
+		name          string
+		network       string
+		networkPrefix int
+		expectErr     bool
```

Comment:
it may be nice to check expected error message, as it may start failing for the wrong reason eventually

Reply:
this is done.